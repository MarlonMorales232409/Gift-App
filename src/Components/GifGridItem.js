import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ title, url }) => {
  return (
    <div className="gift animate__animated animate__bounce animate__fadeIn">
      <img src={url} alt={title} className="gift-image" />
      <p className="gift-title">{title}</p>
    </div>
  );
};

export default GifGridItem;

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
