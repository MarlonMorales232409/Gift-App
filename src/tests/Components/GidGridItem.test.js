import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../Components/GifGridItem";

const title = "Testing";
const url = "https://localhost/img/1";

let wrapper = shallow(<GifGridItem title={title} url={url} />);

describe("testing in GiftGridItem Component", () => {
  test("Must to show the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("it must to have P with the title", () => {
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);
  });

  test("img must to have an src attribute inside", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Div elemment should have the class name animate__fadeIn", () => {
    const div = wrapper.find("div");

    expect(div.hasClass("animate__fadeIn")).toBe(true);
  });
});
