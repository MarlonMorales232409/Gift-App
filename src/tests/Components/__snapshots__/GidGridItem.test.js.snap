// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing in GiftGridItem Component Must to show the component correctly 1`] = `
<div
  className="gift animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="Testing"
    className="gift-image"
    src="https://localhost/img/1"
  />
  <p
    className="gift-title"
  >
    Testing
  </p>
</div>
`;
